profiles/seccomp/kernel_linux.go

```diff
@@ -0,0 +1,83 @@
+package seccomp
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+
+	"golang.org/x/sys/unix"
+)
+
+// kernelVersion holds information about the kernel.
+type kernelVersion struct {
+	kernel uint // Version of the kernel (i.e., the "4" in "4.1.2-generic")
+	major  uint // Major part of the kernel version (i.e., the "1" in "4.1.2-generic")
+}
+
+func (k *kernelVersion) String() string {
+	return fmt.Sprintf("%d.%d", k.kernel, k.major)
+}
+
+var (
+	currentKernelVersion *kernelVersion
+	once                 sync.Once
+)
+
+// getKernelVersion gets the current kernel version.
+func getKernelVersion() (*kernelVersion, error) {
+	var err error
+	once.Do(func() {
+		var uts unix.Utsname
+		err = unix.Uname(&uts)
+		if err != nil {
+			return
+		}
+		// Remove the \x00 from the release for Atoi to parse correctly
+		currentKernelVersion, err = parseRelease(string(uts.Release[:bytes.IndexByte(uts.Release[:], 0)]))
+	})
+	return currentKernelVersion, err
+}
+
+// parseRelease parses a string and creates a kernelVersion based on it.
+func parseRelease(release string) (*kernelVersion, error) {
+	var (
+		version = kernelVersion{}
+		suffix  string
+	)
+
+	// Just make sure we get the "kernel" and "minor" version. Sometimes we have
+	// 3.12.25-gentoo, but sometimes we just have 3.12-1-amd64. We add a bogus
+	// "-any-suffix" suffix to account for anything after "kernel.major" being
+	// optional, and may not be present in the input string.
+	parsed, err := fmt.Sscanf(release+"-any-suffix", "%d.%d%s", &version.kernel, &version.major, &suffix)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse kernel version %q: %w", release, err)
+	}
+	if parsed < 2 {
+		return nil, fmt.Errorf("failed to parse kernel version %q", release)
+	}
+	return &version, nil
+}
+
+// kernelGreaterEqualThan checks if the host's kernel version is greater than, or
+// equal to the given kernel version.
+//
+// Only "Kernel" and "Major" versions are taken into account, which means that
+// 3.12.25-gentoo and 3.12-1-amd64 are considered equal (kernel: 3, major: 12).
+func kernelGreaterEqualThan(v string) (bool, error) {
+	minVersion, err := parseRelease(v)
```

Comment:
I was later thinking that, given that we'd only support `<kernel>.<major>` in the seccomp-profile JSON, I should probably update this to _not_ use  `parseRelease()`, but only accept `<kernel>.<major>`  (i.e., not allow specifying `3.12.25-gentoo` in the JSON file)

Reply:
Updated this, and added separate handling for `minVersion`